vespa/deployment.py

```diff
@@ -1044,41 +1385,59 @@ def _start_prod_deployment(self, disk_folder: str) -> None:
-        canonical_message = "POST" + "\n" + url + "\n" + timestamp + "\n" + content_hash
-        signature = self.api_key.sign(
-            canonical_message.encode("UTF-8"), ec.ECDSA(hashes.SHA256())
+        if self.control_plane_auth_method == "api_key":
+            url = (
+                "https://"
+                + self.connection.host
+                + ":"
+                + str(self.connection.port)
+                + deploy_path
+            )
+            digest = hashes.Hash(hashes.SHA256(), default_backend())
+            digest.update(
+                multipart_data.to_string()
+            )  # This moves the buffer position to the end
+            multipart_data._buffer.seek(
+                0
+            )  # Needs to be reset. Otherwise, no data will be sent
+            content_hash = standard_b64encode(digest.finalize()).decode("UTF-8")
+            timestamp = (
+                datetime.utcnow().isoformat() + "Z"
+            )  # Java's Instant.parse requires the neutral time zone appended
+            canonical_message = (
+                "POST" + "\n" + url + "\n" + timestamp + "\n" + content_hash
+            )
+            signature = self.api_key.sign(
+                canonical_message.encode("UTF-8"), ec.ECDSA(hashes.SHA256())
+            )
+            headers = {
+                "X-Timestamp": timestamp,
+                "X-Content-Hash": content_hash,
+                "X-Key-Id": self.tenant + ":" + self.application + ":" + "default",
+                "X-Key": self.api_public_key_bytes,
+                "X-Authorization": standard_b64encode(signature),
+            }
+        elif self.control_plane_auth_method == "access_token":
+            headers = {
+                "Authorization": "Bearer " + self.control_plane_access_token,
+            }
+        # Read the content of multipart_data into a bytes object
+        multipart_data_bytes: bytes = multipart_data.to_string()
+        headers["Content-Length"] = str(len(multipart_data_bytes))
+        # Update the headers to include the Content-Type
+        headers["Content-Type"] = multipart_data.content_type
+        # Convert multipart_data_bytes to type BytesIO
+        multipart_data_bytes = BytesIO(multipart_data_bytes)
+        response = self._request(
+            "POST", deploy_path, body=multipart_data_bytes, headers=headers
         )
-        headers = {
-            "X-Timestamp": timestamp,
-            "X-Content-Hash": content_hash,
-            "X-Key-Id": self.tenant + ":" + self.application + ":" + "default",
-            "X-Key": self.api_public_key_bytes,
-            "X-Authorization": standard_b64encode(signature),
-            "Content-Type": multipart_data.content_type,
-        }
-
-        response = requests.post(url, data=multipart_data, headers=headers)
-
-        message = response.json()["message"]
+        message = response.get("message", "No message provided")
+        build_no = int(response.get("build", None))
```

Comment:
Note you could read here whether the build was identical to the previous, and if so, only then would you need to poll for the previous deployable one. It could also be nice for the user to see this in a more direct fashion. 

```json
{
  "message": "bla bla",
  "build": 123,
  "skipped": true
}
```